src/server/routes/api.ts

```diff
@@ -732,35 +670,38 @@ connection.once('open', () => {
     // Get a gene list by id
     router.get('/genelist/:id', (req, res, next) => {
         // Return an empty array in case no id was passed or no params
-        if (!req.params || !req.params.id) { res.json({ items: [] }); } else {
-            GenesLinks.find({ geneA_ensembl_gene_id: req.params.id }).exec(async (err, links) => {
-                const arrA = await links.slice().map((slink) => {
-                    return slink.toJSON()['geneB_ensembl_gene_id'];
-                });
-
-                GenesLinks.find({ geneB_ensembl_gene_id: req.params.id }, async (errB, linkB) => {
-                    const arrB = await linkB.slice().map((slink) => {
-                        return slink.toJSON()['geneA_ensembl_gene_id'];
+        if (!req.params || !req.params.id) {
+            res.json({items: []});
+        } else {
+            GenesLinks.find({geneA_ensembl_gene_id: req.params.id}).lean()
+                .exec(async (err, links) => {
+                    const arrA = links.slice().map((slink) => {
+                        return slink.geneB_ensembl_gene_id;
                     });
-                    const arr = [...arrA, ...arrB];
-                    GenesLinks.find({ geneA_ensembl_gene_id: { $in: arr } })
-                        .where('geneB_ensembl_gene_id')
-                        .in(arr)
-                        .exec((errC, linksC) => {
-                            if (errC) {
-                                next(errC);
-                            } else {
-                                const flinks = [...links, ...linkB, ...linksC];
-                                res.setHeader(
-                                    'Cache-Control', 'no-cache, no-store, must-revalidate'
-                                );
-                                res.setHeader('Pragma', 'no-cache');
-                                res.setHeader('Expires', 0);
-                                res.json({ items: flinks });
-                            }
+
+                    GenesLinks.find({geneB_ensembl_gene_id: req.params.id}, async (errB, linkB) => {
```

Comment:
@JessterB I understand that you just changed formatting/added `lean` but if you understand this, can you put a comment? It's not clear to me why we're making up to 3 DB calls as I'd only expect 1 here